src/bios.rs

```diff
@@ -157,10 +159,62 @@ impl Component for Bios {
         crate::component::query_adopt_state()
     }
 
+    // Backup the current grub.cfg and install new static config
+    // - Backup "/boot/loader/grub.cfg" to "/boot/grub2/grub.cfg.bak"
+    // - Remove symlink "/boot/grub2/grub.cfg"
+    // - Install new static grub.cfg
+    // - Remove old "/boot/loader/grub.cfg"
+    fn migrate_static_grub_config(&self, sysroot_path: &str, destdir: &openat::Dir) -> Result<()> {
+        use anyhow::Context;
+        use openat_ext::OpenatDirExt;
+
+        let grub_config_dir = Utf8PathBuf::from(sysroot_path).join("boot/grub2");
+        let grubdir = destdir
+            .sub_dir(grub_config_dir.as_std_path())
+            .context("Opening /boot/grub2")?;
+
+        let grubconfig = grub_config_dir.join(grubconfigs::GRUBCONFIG);
+
+        // On BIOS, /boot/grub2/grub.cfg is symlink to /boot/loader/grub.cfg,
+        // should backup it to /boot/grub2/grub.cfg.bak
+        if !grubconfig.exists() {
+            anyhow::bail!("Not found '{}'", grubconfig);
+        } else if grubconfig.is_symlink() {
+            let realconfig = grubdir.read_link(grubconfigs::GRUBCONFIG)?;
+            let realconfig =
+                Utf8PathBuf::from_path_buf(realconfig).expect("Path should be valid UTF-8");
+            // Resolve symlink location
+            let mut current_config = grub_config_dir.clone();
+            current_config.push(realconfig);
+            let backup_config = grub_config_dir.join(grubconfigs::BACKUP);
+
+            // Backup the current GRUB config which is hopefully working right now
+            println!(
+                "Creating a backup of the current GRUB config '{}' in '{}'...",
+                current_config, backup_config
+            );
+            std::fs::copy(&current_config, &backup_config)
+                .context("Failed to backup GRUB config")?;
+            // Remove the symlink (on BIOS)
+            grubdir.remove_file_optional(grubconfigs::GRUBCONFIG)?;
+            crate::grubconfigs::install(&destdir, None, true)?;
+        } else {
+            anyhow::bail!("'{}' is not a symlink", grubconfig);
+        }
+
+        // Remove the unused file, backuped as /boot/grub2/grub.cfg.bak
+        destdir.remove_file_optional("boot/loader/grub.cfg")?;
```

Comment:
Presumably, this should instead be removing the target of the symlink instead of hardcoding.